### ISSUE TITLE
Segfault for `zabbix_agentd -t` on CentOS 5

### ISSUE THREAD

**Issue by cavaliercoder** [cavaliercoder/libzbxpgsql#13]
Operation seems ok on CentOS 5 x86_64 except when running

    $ zabbix_agentd -t pg.modver

The agent seems to return the module version okay, in amongst a segfault stack dump.

```
zabbix_agentd [6]: Starting agent module libzbxpgsql 0.2.1
zabbix_agentd [6]: ERROR: Got signal [signal:11(SIGSEGV),reason:1,refaddr:0x3]. Crashing ...
zabbix_agentd [6]: ERROR: ====== Fatal information: ======
zabbix_agentd [6]: ERROR: Program counter: 0x3
zabbix_agentd [6]: ERROR: === Registers: ===
zabbix_agentd [6]: ERROR: r8      =     7ffc4bcc5ed0 =      140721580170960 =      140721580170960
zabbix_agentd [6]: ERROR: r9      =          2444258 =             38027864 =             38027864
zabbix_agentd [6]: ERROR: r10     =                0 =                    0 =                    0
zabbix_agentd [6]: ERROR: r11     =         ffffffff =           4294967295 =           4294967295
zabbix_agentd [6]: ERROR: r12     =                0 =                    0 =                    0
pg.modver                                     [s|libzbxpgsql 0.2.1]
zabbix_agentd [6]: ERROR: r13     =     7ffc4bcc6320 =      140721580172064 =      140721580172064
zabbix_agentd [6]: ERROR: r14     =                0 =                    0 =                    0
zabbix_agentd [6]: ERROR: r15     =                0 =                    0 =                    0
zabbix_agentd [6]: ERROR: rdi     =     7f7056311480 =      140120459121792 =      140120459121792
zabbix_agentd [6]: ERROR: rsi     =                0 =                    0 =                    0
zabbix_agentd [6]: ERROR: rbp     =     7ffc4bcc6200 =      140721580171776 =      140721580171776
zabbix_agentd [6]: ERROR: rbx     =          2443400 =             38024192 =             38024192
zabbix_agentd [6]: ERROR: rdx     =                3 =                    3 =                    3
zabbix_agentd [6]: ERROR: rax     =                0 =                    0 =                    0
zabbix_agentd [6]: ERROR: rcx     =                0 =                    0 =                    0
zabbix_agentd [6]: ERROR: rsp     =     7ffc4bcc61e8 =      140721580171752 =      140721580171752
zabbix_agentd [6]: ERROR: rip     =                3 =                    3 =                    3
zabbix_agentd [6]: ERROR: efl     =            10246 =                66118 =                66118
zabbix_agentd [6]: ERROR: csgsfs  =               33 =                   51 =                   51
zabbix_agentd [6]: ERROR: err     =               14 =                   20 =                   20
zabbix_agentd [6]: ERROR: trapno  =                e =                   14 =                   14
zabbix_agentd [6]: ERROR: oldmask =                0 =                    0 =                    0
zabbix_agentd [6]: ERROR: cr2     =                3 =                    3 =                    3
zabbix_agentd [6]: ERROR: === Backtrace: ===
```

**Comment by jamescook000** [comment]
I have come across the same behaviour on rhel/centos 5 and found that changing the compilation optimisation level resolved the agent crashing problem on this platform. By default it was set to '-02' and when removing this option and setting CFLAGS to '-g' the zabbix agent ceases to crash.

**Comment by jamescook000** [comment]
Please also see as this may help : https://support.zabbix.com/browse/ZBX-8470

**Comment by cavaliercoder** [comment]
Confirmed fix in Zabbix v2.5+ as per https://support.zabbix.com/browse/ZBX-8470
